catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/main/java/org/codice/ddf/catalog/subscriptionstore/common/SubscriptionPersistor.java

```diff
@@ -0,0 +1,171 @@
+ * of the service.
+ */
+public class SubscriptionPersistor {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionPersistor.class);
+
+    private static final String SUBSCRIPTION_ID_SOLR = "id";
+
+    private static final String SUBSCRIPTION_TYPE = "type";
+
+    private static final String SUBSCRIPTION_SERIALIZED_FILTER = "filter";
+
+    private static final String SUBSCRIPTION_CALLBACK_URL = "callback";
+
+    private final PersistentStore persistentStore;
+
+    public SubscriptionPersistor(PersistentStore persistentStore) {
+        this.persistentStore = persistentStore;
+    }
+
+    /**
+     * Get all the subscription metadata objects in the persistent store, regardless of type.
+     *
+     * @return a map of subscription id's to their accompanying metadata.
+     * @throws SubscriptionStoreException if a problem occurs during the read.
+     */
+    public Map<String, SubscriptionMetadata> getSubscriptions() {
+        List<Map<String, Object>> results;
+        try {
+            results = persistentStore.get(PersistentStore.EVENT_SUBSCRIPTIONS_TYPE);
+        } catch (PersistenceException e) {
+            throw new SubscriptionStoreException("Exception while reading subscriptions from Solr: ",
+                    e);
+        }
+
+        return results.stream()
+                .map(this::formatIfPersistentItem)
+                .map(this::mapToMetadata)
+                .collect(Collectors.toMap(SubscriptionMetadata::getId, Function.identity()));
+    }
+
+    /**
+     * Add the subscription metadata to the persistent store using it's id for the document key.
+     * <p>
+     * Insertion operations with the same key overwrite the previous value, which means a duplicate
+     * being added can be treated as a no-op, minus the network overhead.
+     *
+     * @param metadata the subscription metadata to add to the persistent store.
+     * @throws SubscriptionStoreException if a problem occurs during insert.
+     */
+    public void insert(SubscriptionMetadata metadata) {
+        LOGGER.debug("Adding [{}] to persistence store. ", metadata.getId());
+        PersistentItem persistentSubscription = metadataToPersistentItem(metadata);
+        try {
+            persistentStore.add(PersistentStore.EVENT_SUBSCRIPTIONS_TYPE, persistentSubscription);
+        } catch (PersistenceException e) {
+            throw new SubscriptionStoreException("Exception while persisting subscription: ", e);
+        }
+    }
+
+    /**
+     * Remove the subscription metadata from the persistent store that corresponds to the given id.
+     * <p>
+     * Redundant deletes are valid as a no-op to support javax.cache events propogating write-through
+     * processing to other cache nodes.
+     *
+     * @param subscriptionId the unique id of the subscription to delete.
+     * @throws SubscriptionStoreException if a problem occurs during delete.
+     */
+    public void delete(String subscriptionId) {
```

Comment:
I'm imagining this will be used in future iterations since it is only in unit tests currently. just clarifying.